src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -538,4 +538,38 @@ describe('ReactServerRendering', function() {
     var markup = ReactServerRendering.renderToStaticMarkup(<Baz />);
     expect(markup).toBe('<div></div>');
   });
+
+  it('warns when children are mutated before render', function() {
+    function normalizeCodeLocInfo(str) {
+      return str.replace(/\(at .+?:\d+\)/g, '(at **)');
+    }
+
+    spyOn(console, 'error');
+    var children = [<span key={0} />, <span key={1} />, <span key={2} />];
+    var element = <div>{children}</div>;
+    children[1] = <p key={1} />; // Mutation is illegal
+    ReactServerRendering.renderToString(element);
+    expect(console.error.calls.count()).toBe(1);
+    expect(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(
+      'Warning: Component\'s children should not be mutated.\n    in div (at **)'
+    );
+  });
+
+  it('should warn when children are mutated', function() {
```

Comment:
Nit: not very clear why there are two tests, and how they are different (right not the name only differences is starting with "should")